t/04-patch/02-ngx-read-body-block.t

```diff
@@ -0,0 +1,49 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use Test::Nginx::Socket 'no_plan';
+
+repeat_each(2);
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: ngx.req.read_body() should work for HTTP2 GET requests that doesn't carry the content-length header
+--- config
+    location = /test {
+        content_by_lua_block {
+            local ok, err = pcall(ngx.req.read_body)
+            ngx.say(ok, " err: ", err)
+        }
+    }
+--- http2
+--- request
+GET /test
+hello, world
+--- more_headers
+Content-Length:
```

Comment:
Are these two lines of code required? A simple GET request would be like this in a text manner:

```
GET / HTTP/1.1
Host: example.com
```

I think removing these two lines of code would make it simulate the simple GET request in HTTP 2.0? Thought?

Reply:
I did not get what is the problem of adding these two lines.

Reply:
It seems that whether it is mandatory or not will not affect my version of  `curl`, but we cannot guarantee that all `curl` will behave this way, so adding this is a safe behavior.